### ISSUE TITLE
Null safety on locale

### ISSUE THREAD

**Issue by masimun** [Tienisto/flutter-fast-i18n#11]
When compiling I get the following error: 
```
/D:/flutter/.pub-cache/hosted/pub.dartlang.org/fast_i18n-3.0.3/lib/fast_i18n.dart:13:46: Error: Method 'toLanguageTag' cannot be called on 'Locale?' because it is potentially null.
 - 'Locale' is from 'dart:ui'.
Try calling using ?. instead.
      WidgetsBinding.instance?.window.locale.toLanguageTag();
```

The issue is resolved modifying:
```diff
class FastI18n {
  static const _localePartsDelimiter = '-';

  /// Returns the locale string used by the device.
  static String? getDeviceLocale() =>
-      WidgetsBinding.instance?.window.locale.toLanguageTag();
+     WidgetsBinding.instance?.window.locale?.toLanguageTag();

  /// Returns the candidate (or part of it) if it is supported.
  /// Fallbacks to base locale.
  /// Please note that locales are case sensitive.
  static String selectLocale(
      String candidate, List<String> supported, String baseLocale) {
    // normalize
    candidate = Utils.normalize(candidate);
```

**My config is:**
$ cat pubspec.yaml
```
environment:
    sdk: '>=2.12.0 <3.0.0'
```

$ flutter doctor
```
[√] Flutter (Channel stable, 2.0.4, on Microsoft Windows [Version 10.0.19042.867], locale en-BE)
```

Is it possible to verify this issue?

**Comment by masimun** [comment]
> You use Flutter Web, right?

Yes indeed

**Comment by Tienisto** [comment]
You can use the dev version `3.0.4-dev.0` for now.
It seems to be fixed in https://github.com/flutter/flutter/issues/79555. But somehow it has not landed in Flutter 2.0.4 yet.

**Comment by masimun** [comment]
Running:
```
$ flutter channel master
$ flutter upgrade
```
Seems to resolve the issue for now
```
[√] Flutter (Channel master, 2.1.0-13.0.pre.550, on Microsoft Windows [Version 10.0.19042.867], locale en-BE)
```
